src/classes/TestResult.js

```diff
@@ -142,24 +142,28 @@ export default class TestResult extends BubblingEventTarget {
 
 		this.tests = this.test.tests?.map(t => new TestResult(t, this, childOptions));
 
-		delay(1)
+		// Promise resolved when all tests (including nested) are finished
+		this.allFinished = delay(1)
 			.then(() => this.test.beforeAll?.())
 			.then(() => {
 				if (this.test.isTest) {
 					if (this.test.skip) {
-						this.skip();
+						return this.skip();
```

Comment:
If we're skipping a test, its before/after hooks should also be skipped.

Reply:
Done